src/foam/lib/json/StringParser.java

```diff
@@ -36,19 +42,21 @@ public StringParser() {
 
   public PStream parse(PStream ps, ParserContext x) {
     if ( ! ps.valid() ) return null;
-    char delim = ps.head();
 
-    if ( delim != '"' && delim != '\'' ) return null;
-
-    ps = ps.tail();
-    char lastc = delim;
+    ps = ps.apply(delimiterParser, x);
+    if ( ps == null ) return null;
+    Parser delimiter = new Literal((String)ps.value());
+    boolean isMultiLine = (ps.value().equals("\"\"\"")) ? true : false;
 
     StringBuilder builder = sb.get();
+    PStream result;
+    char lastc = ps.head();
 
     while ( ps.valid() ) {
       char c = ps.head();
 
-      if ( c == delim && lastc != ESCAPE ) break;
+      result = ps.apply(delimiter, x);
+      if ( result != null && lastc != ESCAPE )  break;
```

Comment:
Instead of isMultiline, you can just do `ps = result` here, and then `break`  Then outside of the loop we can just do `return ps.setValue(builder.toString());` because the delimiter parser will have consumed PStream till after the delimiter regardless of what it was.

Reply:
I think you can also clean up this logic a bit.  Only attempt to parse the delimiter if `lastc != ESCAPE` no use checking if the next characters are the string delimiter if we know we're going to ignore the result.

Reply:
im not sure what you mean by this